pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -1901,7 +1901,13 @@ public String getString(int columnIndex) throws SQLException {
       Field field = fields[columnIndex - 1];
       Object obj = internalGetObject(columnIndex, field);
       if (obj == null) {
-        return null;
+        // internalGetObject() knows jdbc-types and some extra like hstore. It does not know of
+        // PGobject based types like geometric types but getObject does
+        obj = getObject(columnIndex);
+        if (obj == null) {
+          return null;
+        }
+        return getObject(columnIndex).toString();
```

Comment:
This does look odd: why is it calling `getObject` twice?

Reply:
Well spotted - corrected